mono/mini/aot-compiler.c

```diff
@@ -8990,6 +8993,9 @@ emit_llvm_file (MonoAotCompile *acfg)
 #else
 		opts = g_strdup ("-targetlibinfo -no-aa -basicaa -notti -instcombine -simplifycfg -inline-cost -inline -sroa -domtree -early-cse -lazy-value-info -correlated-propagation -simplifycfg -instcombine -simplifycfg -reassociate -domtree -loops -loop-simplify -lcssa -loop-rotate -licm -lcssa -loop-unswitch -instcombine -scalar-evolution -loop-simplify -lcssa -indvars -loop-idiom -loop-deletion -loop-unroll -memdep -gvn -memdep -memcpyopt -sccp -instcombine -lazy-value-info -correlated-propagation -domtree -memdep -adce -simplifycfg -instcombine -strip-dead-prototypes -domtree -verify");
 #endif
+	if (acfg->llvm_opts != NULL)
```

Comment:
This will leak memory, and will override the default one. If overriding is the expected behavior, we should put lines 8987 to 8995 in the else branch of this `acfg->llvm_opts != NULL`

Reply:
Well, I could point out that this is a tiny leak on a one-off build, this is not a long running process.

Anyways, patch attached.